integrations/go/workload_executor_test.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/integration/mtest"
+	"go.mongodb.org/mongo-driver/mongo/integration/unified"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+)
+
+// hasLoop returns whether or not testCase tc contains a loop operation
+func hasLoop(tc *unified.TestCase) bool {
+	if tc == nil {
+		return false
+	}
+	for _, op := range tc.Operations {
```

Comment:
No need to change. Though AFAICT it looks like the definition of [loop](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#loop) does not prohibit it appearing as a nested operation within a [withTransaction](https://github.com/mongodb/specifications/tree/master/source/transactions-convenient-api/tests#withtransaction-operation). IMO it does not seem worth adding the extra logic to check for that. I don't think there is a realistic use case for looping within a single transaction.